actors/miner/src/lib.rs

```diff
@@ -4935,7 +4973,17 @@ fn activate_new_sector_infos(
     data_activations: Vec<DataActivationOutput>,
     pledge_inputs: &NetworkPledgeInputs,
     info: &MinerInfo,
+    emit_event: bool,
```

Comment:
@anorth We need this as we only want the emit here if this call is from the cron callback to `CONFIRM_SECTOR_PROOFS_VALID` which in turn is scheduled by the vanilla prove commit method.

For the prove commit aggregate method, we don't want to emit the event here as we also need to check if we have enough funds for gas fees. Doing it this way makes it work nicely with the tests.

Reply:
I don't think this is the right trade-off. Emitting the events here is a good place to ensure that future code changes will still emit events exactly according to activations, and will simplify code elsewhere. There are more paths into sector activation coming in `integration/direct-onboarding`.

We don't need to make checks about funds etc before emitting events - if those checks fail the whole thing will abort anyway. We may need to update testing support code to allow us to write the main code in the straightforward way.

Reply:
This is done. We no longer have this flag.